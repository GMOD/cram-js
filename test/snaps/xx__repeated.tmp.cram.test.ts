import { testFile } from '../testUtil'
testFile('xx#repeated.tmp.cram')
