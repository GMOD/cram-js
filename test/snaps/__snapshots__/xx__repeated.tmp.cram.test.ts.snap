// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole xx#repeated.tmp.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 44,
      "_size": 18,
      "alignmentSpan": 0,
      "crc32": 3519680774,
      "landmarks": [
        0,
        119,
      ],
      "length": 187,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 44,
        "_size": 18,
        "alignmentSpan": 0,
        "crc32": 3519680774,
        "landmarks": [
          0,
          119,
        ],
        "length": 187,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/Users/rbuels/dev/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": false,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 6,
          "parameters": {
            "length": 4,
            "offset": 0,
          },
          "parametersBytes": 2,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "CF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 16,
          },
          "parametersBytes": 1,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              10,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 249,
      "_size": 18,
      "alignmentSpan": 20,
      "crc32": 1336459378,
      "landmarks": [
        143,
      ],
      "length": 351,
      "numBases": 60,
      "numBlocks": 7,
      "numLandmarks": 1,
      "numRecords": 6,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 249,
        "_size": 18,
        "alignmentSpan": 20,
        "crc32": 1336459378,
        "landmarks": [
          143,
        ],
        "length": 351,
        "numBases": 60,
        "numBlocks": 7,
        "numLandmarks": 1,
        "numRecords": 6,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 67,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "readName": "S",
              "sequenceId": 0,
              "uniqueId": 399,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "S",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 398,
          },
          {
            "alignmentStart": 11,
            "cramFlags": 5,
            "flags": 131,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "S",
              "sequenceId": 0,
              "uniqueId": 400,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "S",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 399,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 67,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "readName": "S",
              "sequenceId": 0,
              "uniqueId": 401,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "S",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 400,
          },
          {
            "alignmentStart": 11,
            "cramFlags": 5,
            "flags": 131,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "S",
              "sequenceId": 0,
              "uniqueId": 402,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "S",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 401,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 67,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "readName": "S",
              "sequenceId": 0,
              "uniqueId": 403,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "S",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 402,
          },
          {
            "alignmentStart": 11,
            "cramFlags": 1,
            "flags": 131,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "S",
              "sequenceId": 0,
              "uniqueId": 402,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "S",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 403,
          },
        ],
        "header": {
          "_endPosition": 433,
          "_size": 36,
          "compressedSize": 32,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            20,
            6,
            0,
            5,
            4,
            11,
            12,
            15,
            16,
            255,
            255,
            255,
            255,
            15,
            187,
            244,
            222,
            109,
            132,
            151,
            161,
            25,
            221,
            166,
            224,
            116,
            82,
            22,
            67,
            220,
          ],
          "contentId": 0,
          "contentPosition": 397,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 2151937019,
          "parsedContent": {
            "_endPosition": 281,
            "_size": 32,
            "contentIds": [
              11,
              12,
              15,
              16,
            ],
            "md5": [
              187,
              244,
              222,
              109,
              132,
              151,
              161,
              25,
              221,
              166,
              224,
              116,
              82,
              22,
              67,
              220,
            ],
            "numBlocks": 5,
            "numContentIds": 4,
            "numRecords": 6,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 20,
            "refSeqStart": 1,
          },
          "uncompressedSize": 32,
        },
      },
    ],
  },
]
`;
